### PR TITLE
[13.0][IMP] account_financial_report: Add tax detail to the general ledger report

### PR DESCRIPTION
**Pull request by victoralmau**
Changes done in general ledger:
- [x] Add tax detail to the general ledger report.
- [x] Refactor code to add grouping
- [x] Add `tax_line_id` field to taxes column in general ledger.

Please @pedrobaeza and @sergio-teruel can you review it?

@Tecnativa TT38721

### CODE REVIEW

File: account_financial_report/report/general_ledger.py [hunk 1]
```diff
@@ -50,10 +50,13 @@ def _get_taxes_data(self, taxes_ids):
...
 
-    def _get_acc_prt_accounts_ids(self, company_id):
+    def _get_acc_prt_accounts_ids(self, company_id, grouped_by):
+        internal_types = ["receivable", "payable"]
+        if grouped_by == "taxes":
+            internal_types.append("other")
         accounts_domain = [
             ("company_id", "=", company_id),
-            ("internal_type", "in", ["receivable", "payable"]),
+            ("internal_type", "in", internal_types),
```
Comment (pedrobaeza) [inline account_financial_report/report/general_ledger.py]: For taxes, the internal types `receivable` and `payable` mustn't apply.

File: account_financial_report/wizard/general_ledger_wizard_view.xml [hunk 2]
```diff
@@ -22,11 +22,11 @@
                             <field name="date_from" />
                             <field name="date_to" />
                             <field name="fy_start_date" invisible="1" />
+                            <field name="grouped_by" />
```
Comment (pedrobaeza) [inline account_financial_report/wizard/general_ledger_wizard_view.xml]: Please keep this in the other column. You can move `target_move` to the first column that feels more natural.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
It still puts "Partner initial balance" and "Partner ending balance".

**Comment by victoralmau** [comment]
Changes done.

**Comment by pedrobaeza** [comment]
For the accounts not taken into account for taxes it still puts "Tax initial balance":

Selección_002

Please change such text.

**Comment by victoralmau** [comment]
> For the accounts not taken into account for taxes it still puts "Tax initial balance":
> 
> Selección_002
> 
> Please change such text.

Ok, changes done.

**Review by pedrobaeza — APPROVED** [review]
Great new feature!

**Review by carlosdauden — APPROVED** [review]

**Comment by MiquelRForgeFlow** [comment]
Forward ports?

**Comment by pedrobaeza** [comment]
@MiquelRForgeFlow you can proceed if you want with them. If not, we will do it on the next free slot.

**Comment by victoralmau** [comment]
> Forward ports?

Hi, in a few hours I will make it to v14 and v15.

**Pull request merged**